my_game.py

```diff
@@ -307,10 +329,12 @@ def on_update(self, delta_time):
             self.game_over()
 
         # Check if player dies when touching "deadly" tile
-        for deadly_tile in self.map.sprite_lists["deadly"]:
-            if deadly_tile.collides_with_point(self.player.position):
-                self.player.lives -= 1
-                self.reset()
+        # Only check if player does not ride something
```

Comment:
bacause deadly tile could be water, but the flayer is riding an object to avoid the water

Reply:
flayer